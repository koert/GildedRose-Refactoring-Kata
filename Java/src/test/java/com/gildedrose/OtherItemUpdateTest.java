package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

/**
 * @author Koert Zeilstra
 */
class OtherItemUpdateTest {

  public static final String NAME = "Some other item";
  public static final Integer[] SELL_IN = {-1, 0, 4, 5, 6, 9, 10, 11};
  public static final Integer[] QUALITY = {0, 10, 50};

  OtherItemUpdate update = new OtherItemUpdate();

  @DisplayName("update")
  @Test
  void update() {
    CombinationApprovals.verifyAllCombinations(this::doUpdateQuality, SELL_IN, QUALITY);
  }

  String doUpdateQuality(int sellIn, int quality) {
    Item item = new Item(NAME, sellIn, quality);    
    update.update(item);
    return item.toString();
  }

}