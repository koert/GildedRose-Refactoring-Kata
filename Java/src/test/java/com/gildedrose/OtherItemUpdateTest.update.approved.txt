[-1, 0] => Some other item, -2, 0 
[-1, 10] => Some other item, -2, 8 
[-1, 50] => Some other item, -2, 48 
[0, 0] => Some other item, -1, 0 
[0, 10] => Some other item, -1, 8 
[0, 50] => Some other item, -1, 48 
[4, 0] => Some other item, 3, 0 
[4, 10] => Some other item, 3, 9 
[4, 50] => Some other item, 3, 49 
[5, 0] => Some other item, 4, 0 
[5, 10] => Some other item, 4, 9 
[5, 50] => Some other item, 4, 49 
[6, 0] => Some other item, 5, 0 
[6, 10] => Some other item, 5, 9 
[6, 50] => Some other item, 5, 49 
[9, 0] => Some other item, 8, 0 
[9, 10] => Some other item, 8, 9 
[9, 50] => Some other item, 8, 49 
[10, 0] => Some other item, 9, 0 
[10, 10] => Some other item, 9, 9 
[10, 50] => Some other item, 9, 49 
[11, 0] => Some other item, 10, 0 
[11, 10] => Some other item, 10, 9 
[11, 50] => Some other item, 10, 49 
